library/alloc/src/collections/linked_list.rs

```diff
@@ -149,22 +159,25 @@ impl<T> Node<T> {
         Node { next: None, prev: None, element }
     }
 
-    fn into_element(self: Box<Self>) -> T {
+    fn into_element<A: Allocator>(self: Box<Self, A>) -> T {
         self.element
     }
 }
 
 // private methods
-impl<T> LinkedList<T> {
+impl<T, A: Allocator> LinkedList<T, A> {
     /// Adds the given node to the front of the list.
+    ///
+    /// # Safety
+    /// `node` must point to a valid node that was boxed using the list's allocator.
     #[inline]
-    fn push_front_node(&mut self, mut node: Box<Node<T>>) {
+    unsafe fn push_front_node(&mut self, node: Unique<Node<T>>) {
```

Comment:
Ah, we were actually quite happy that `Unique` is basically unused in the standard library. It'd be better to avoid introducing new uses of this type until its status is clarified. Currently it makes absolutely no difference to `NonNull`. There are some vague plans to maybe attach alias information to that type, but currently this does not happen. (Cc https://github.com/rust-lang/unsafe-code-guidelines/issues/384)

Why was the type chosen here?

Reply:
`Unique` seemed like a better fit since the method takes ownership of the node. Changing the signature to use `NonNull` should be fine; the doc comment should probably also mention that the pointer shouldn’t be used again.

Reply:
`Unique` is `Copy`, it can't really indicate ownership.